numba/core/interpreter.py

```diff
@@ -80,21 +81,24 @@ def get_assignment_source(self, destname):
         return None
 
 
-def _remove_assignment_definition(old_body, idx, func_ir):
+def _remove_assignment_definition(old_body, idx, func_ir, ignore_missing=False):
```

Comment:
I think ideally this should involve tracking what has previously been removed opposed to ignoring duplicate removals? It seems like `ignore_missing=True` essentially aliases "Remove a previously existing valid `rhs` from `lhs`'s definitions" and "Remove a non-sensical `rhs` (i.e. from a bug elsewhere!) from `lhs`'s definitions".

The question here is really... what's the risk? Is the only way to trigger such an issue via a bug in the IR state from else where? If so, leave as is? What do you think @njriasan?

Reply:
@stuartarchibald I've pushed a change that keeps a dictionary of which variables have been deleted. Let me know if this changes looks sufficient to you.

Reply:
> I don't know if there is a realistic risk, but I suggestion is reasonable and also don't very difficult to implement. Since in the past I found issues with other peepholes in the definitions and this stuff is done manually, I think its worth effort to add a set of deleted definitions.

OK, if the effort is short then please feel free to add this, many thanks. I guess having this also helps with ensuring self-consistency as it's enforcing a prescribed behaviour, it'll probably be more easily debugged as a result too.